indispensable/src/commonTest/kotlin/PemTest.kt

```diff
@@ -0,0 +1,55 @@
+import at.asitplus.signum.indispensable.CryptoPrivateKey
+import at.asitplus.signum.indispensable.asn1.decodeFromPem
+import at.asitplus.signum.indispensable.asn1.encodeToPEM
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeInstanceOf
+
+@OptIn(ExperimentalStdlibApi::class)
+class PemTest : FreeSpec({
+    "SEC1" {
+        val sec1 = """
+            -----BEGIN EC PRIVATE KEY-----
+            MHcCAQEEIGwHU3LKj2fCxiUWB76jCnxIOJ2KAgYKbYGays8h/g+goAoGCCqGSM49
+            AwEHoUQDQgAET+Zr8vrF+kdr1zpjK3ufUv1fd7DS0s8Yf8/Ny3Hb4I57Sz20Zabp
+            brDmqFB7AmrWhdejOPHn9+Ln51i42bCdGQ==
+            -----END EC PRIVATE KEY-----
+        """.trimIndent()
+
+        CryptoPrivateKey.EC.decodeFromPem(sec1).getOrThrow().pemEncodeSec1().getOrThrow() shouldBe sec1
+        (CryptoPrivateKey.decodeFromPem(sec1).getOrThrow() as CryptoPrivateKey.EC).pemEncodeSec1()
+            .getOrThrow() shouldBe sec1
+    }
+
+    "PKCS#8" {
```

Comment:
Maybe also test for expected failures, i.e. strings that can not be parsed correctly

Reply:
see above